src/lib/sql/sql_translator.cpp

```diff
@@ -29,76 +48,103 @@
-          {PredicateCondition::LessThan, PredicateCondition::GreaterThan},
-          {PredicateCondition::GreaterThanEquals, PredicateCondition::LessThanEquals},
-          {PredicateCondition::LessThanEquals, PredicateCondition::GreaterThanEquals}};
-
-  auto it = predicate_condition_for_reverse_order.find(predicate_condition);
-  if (it != predicate_condition_for_reverse_order.end()) {
-    return it->second;
-  }
+/**
+ * Is the expression a predicate that our Join Operators can process directly?
+ * That is, is it of the form <column> <predicate_condition> <column>?
+ */
+bool is_trivial_join_predicate(const AbstractExpression& expression, const AbstractLQPNode& left_input,
+                               const AbstractLQPNode& right_input) {
+  if (expression.type != ExpressionType::Predicate) return false;
+
+  const auto* binary_predicate_expression = dynamic_cast<const BinaryPredicateExpression*>(&expression);
+  if (!binary_predicate_expression) return false;
 
-  return predicate_condition;
+  const auto left_in_left = left_input.find_column_id(*binary_predicate_expression->left_operand());
+  const auto right_in_right = right_input.find_column_id(*binary_predicate_expression->right_operand());
+  const auto right_in_left = left_input.find_column_id(*binary_predicate_expression->right_operand());
+  const auto left_in_right = right_input.find_column_id(*binary_predicate_expression->left_operand());
+
+  return (left_in_left && right_in_right) || (right_in_left && left_in_right);
 }
+}  // namespace
 
-JoinMode translate_join_type_to_join_mode(const hsql::JoinType join_type) {
-  static const std::unordered_map<const hsql::JoinType, const JoinMode> join_type_to_mode = {
-      {hsql::kJoinInner, JoinMode::Inner}, {hsql::kJoinFull, JoinMode::Outer},      {hsql::kJoinLeft, JoinMode::Left},
-      {hsql::kJoinRight, JoinMode::Right}, {hsql::kJoinNatural, JoinMode::Natural}, {hsql::kJoinCross, JoinMode::Cross},
-  };
+namespace opossum {
 
-  auto it = join_type_to_mode.find(join_type);
-  DebugAssert(it != join_type_to_mode.end(), "Unable to handle join type.");
-  return it->second;
+SQLTranslator::SQLTranslator(const UseMvcc use_mvcc,
+                             const std::shared_ptr<SQLIdentifierContextProxy>& external_sql_identifier_context_proxy,
+                             const std::shared_ptr<ParameterIDAllocator>& parameter_id_allocator)
+    : _use_mvcc(use_mvcc),
+      _external_sql_identifier_context_proxy(external_sql_identifier_context_proxy),
+      _parameter_id_allocator(parameter_id_allocator) {}
+
+const std::unordered_map<ValuePlaceholderID, ParameterID>& SQLTranslator::value_placeholders() const {
+  return _parameter_id_allocator->value_placeholders();
+}
+
+std::shared_ptr<SQLIdentifierContext> SQLTranslator::sql_identifier_context() const { return _sql_identifier_context; }
+
+std::vector<std::shared_ptr<AbstractLQPNode>> SQLTranslator::translate_sql(const std::string& sql) {
```

Comment:
I don't like that the SQLTranslator now takes either an already parsed string or the string itself. Why doesn't the old approach work anymore?

Reply:
I think this method is mostly convenience, I will remove it